src/Compilers/CSharp/Portable/Binder/PatternVariableFinder.cs

```diff
@@ -165,6 +165,17 @@ public override void VisitArgument(ArgumentSyntax node)
                 return;
             }
 
+            switch (node.Identifier.Parent.Parent.Parent.Kind())
```

Comment:
This expression "node.Identifier.Parent.Parent.Parent" is mysterious. A comment to explain would be helpful.
What happens if an "out var" is given as an indexer argument?